{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This paper looked at a study where bone marrow was taken from 4 healthy patients (BM1-BM4), 3 patients with thalassemia major (BT1-BT3) and 5 patients with sickle cell disorder (SCD1-SC5). From this bone marrow, the CD34+ cells were analyzed using 10X chromium platform. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Also, this random snippet of a quote: \"a Lineage depleted CD34+ cells from one normal sample.\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "27-Nov-2024 12:32:47 DEBUG utils - Directory ./ already exists. Skipping.\n",
      "27-Nov-2024 12:32:47 INFO GEOparse - File already exist: using local version.\n",
      "27-Nov-2024 12:32:47 INFO GEOparse - Parsing ./GSE133181_family.soft.gz: \n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SERIES: GSE133181\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - PLATFORM: GPL24676\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901485\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901486\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901487\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901488\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901489\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901490\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901491\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901492\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901493\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901494\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901495\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901496\n",
      "27-Nov-2024 12:32:47 DEBUG GEOparse - SAMPLE: GSM3901498\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "basic info about data set:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'title': [\"Single cell transcriptome analysis of normal, Thalassemia and sickle cell donors' bone marrow CD34+ cells\"],\n",
       " 'geo_accession': ['GSE133181'],\n",
       " 'status': ['Public on Oct 01 2019'],\n",
       " 'submission_date': ['Jun 22 2019'],\n",
       " 'last_update_date': ['Dec 31 2019'],\n",
       " 'pubmed_id': ['31697810'],\n",
       " 'web_link': ['https://doi.org/10.1182/blood.2019002301'],\n",
       " 'summary': [\"4 normal, 3 thalassemia major and 5 sickle cell donors' bone marrow CD34+ cells were analyzed using 10X chromium platform and also Lineage depleted CD34+ cells from one normal sample\"],\n",
       " 'overall_design': [\"4 normal, 3 thalassemia major and 5 sickle cell donors' bone marrow CD34+ cells were analyzed\"],\n",
       " 'type': ['Expression profiling by high throughput sequencing'],\n",
       " 'contributor': ['Peng,,Hua'],\n",
       " 'sample_id': ['GSM3901485',\n",
       "  'GSM3901486',\n",
       "  'GSM3901487',\n",
       "  'GSM3901488',\n",
       "  'GSM3901489',\n",
       "  'GSM3901490',\n",
       "  'GSM3901491',\n",
       "  'GSM3901492',\n",
       "  'GSM3901493',\n",
       "  'GSM3901494',\n",
       "  'GSM3901495',\n",
       "  'GSM3901496',\n",
       "  'GSM3901498'],\n",
       " 'contact_name': ['Peng,,Hua'],\n",
       " 'contact_email': ['penghua@njmu.edu.cn'],\n",
       " 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'],\n",
       " 'contact_institute': ['Nanjing Medical University'],\n",
       " 'contact_address': ['101 Longmiandadao'],\n",
       " 'contact_city': ['Nanjing'],\n",
       " 'contact_state': ['Jiangsu'],\n",
       " 'contact_zip/postal_code': ['211116'],\n",
       " 'contact_country': ['China'],\n",
       " 'supplementary_file': ['ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE133nnn/GSE133181/suppl/GSE133181_RAW.tar'],\n",
       " 'platform_id': ['GPL24676'],\n",
       " 'platform_taxid': ['9606'],\n",
       " 'sample_taxid': ['9606'],\n",
       " 'relation': ['BioProject: https://www.ncbi.nlm.nih.gov/bioproject/PRJNA550205',\n",
       "  'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRP211743']}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GSM3901485\n",
      "<built-in method title of str object at 0x0000022C75AD4BF0>\n",
      "{'title': ['BM1'], 'geo_accession': ['GSM3901485'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"normal donor's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: normal donor', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111255', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261304'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901485/suppl/GSM3901485_BM1.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901486\n",
      "<built-in method title of str object at 0x0000022C75AD4FB0>\n",
      "{'title': ['BM2'], 'geo_accession': ['GSM3901486'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"normal donor's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: normal donor', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111254', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261305'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901486/suppl/GSM3901486_BM2.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901487\n",
      "<built-in method title of str object at 0x0000022C75C487F0>\n",
      "{'title': ['BM3'], 'geo_accession': ['GSM3901487'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"normal donor's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: normal donor', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111253', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261306'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901487/suppl/GSM3901487_BM3.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901488\n",
      "<built-in method title of str object at 0x0000022C75DED970>\n",
      "{'title': ['BM4'], 'geo_accession': ['GSM3901488'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"normal donor's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: normal donor', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111252', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261307'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901488/suppl/GSM3901488_BM4.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901489\n",
      "<built-in method title of str object at 0x0000022C75E36870>\n",
      "{'title': ['BT1'], 'geo_accession': ['GSM3901489'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"Major Thalassemia patient's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: Major Thalassemia patient', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111251', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261308'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901489/suppl/GSM3901489_BT1.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901490\n",
      "<built-in method title of str object at 0x0000022C75E403F0>\n",
      "{'title': ['BT2'], 'geo_accession': ['GSM3901490'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"Major Thalassemia patient's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: Major Thalassemia patient', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111250', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261309'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901490/suppl/GSM3901490_BT2.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901491\n",
      "<built-in method title of str object at 0x0000022C75E41F30>\n",
      "{'title': ['BT3'], 'geo_accession': ['GSM3901491'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"Major Thalassemia patient's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: Major Thalassemia patient', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111249', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261310'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901491/suppl/GSM3901491_BT3.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901492\n",
      "<built-in method title of str object at 0x0000022C75E43830>\n",
      "{'title': ['C1'], 'geo_accession': ['GSM3901492'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"normal donor's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: normal donor', 'tissue: bone marrow', 'cell type: Lin-CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111248', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261311'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901492/suppl/GSM3901492_C1.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901493\n",
      "<built-in method title of str object at 0x0000022C75E49570>\n",
      "{'title': ['SCD1'], 'geo_accession': ['GSM3901493'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"sickle cell disease patient's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: sickle cell disease patient', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111247', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261312'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901493/suppl/GSM3901493_SCD1.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901494\n",
      "<built-in method title of str object at 0x0000022C75E4B0B0>\n",
      "{'title': ['SCD2'], 'geo_accession': ['GSM3901494'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"sickle cell disease patient's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: sickle cell disease patient', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111246', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261313'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901494/suppl/GSM3901494_SCD2.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901495\n",
      "<built-in method title of str object at 0x0000022C75E509B0>\n",
      "{'title': ['SCD3'], 'geo_accession': ['GSM3901495'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"sickle cell disease patient's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: sickle cell disease patient', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111245', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261314'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901495/suppl/GSM3901495_SCD3.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901496\n",
      "<built-in method title of str object at 0x0000022C75E526B0>\n",
      "{'title': ['SCD4'], 'geo_accession': ['GSM3901496'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"sickle cell disease patient's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: sickle cell disease patient', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111244', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261315'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901496/suppl/GSM3901496_SCD4.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n",
      "GSM3901498\n",
      "<built-in method title of str object at 0x0000022C75E58370>\n",
      "{'title': ['SCD5'], 'geo_accession': ['GSM3901498'], 'status': ['Public on Oct 01 2019'], 'submission_date': ['Jun 22 2019'], 'last_update_date': ['Oct 02 2019'], 'type': ['SRA'], 'channel_count': ['1'], 'source_name_ch1': [\"sickle cell disease patient's bone marrow cells\"], 'organism_ch1': ['Homo sapiens'], 'taxid_ch1': ['9606'], 'characteristics_ch1': ['individual: sickle cell disease patient', 'tissue: bone marrow', 'cell type: CD34+ cells'], 'treatment_protocol_ch1': ['no treatment'], 'growth_protocol_ch1': ['uncultured'], 'molecule_ch1': ['total RNA'], 'extract_protocol_ch1': [\"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\"], 'data_processing': ['*I1_001.fastq.gz: contains the sample index.', '*R1_001.fastq.gz: contains the cell barcode (first 16 nt) and UMI (next 10 nt).', '*R2_001.fastq.gz: contains the RNA sequence.', \"Single-cell RNA sequencing libraries were prepared using Chromium Single Cell 3’ Library & Gel Bead Kit v2 (10XGenomics, cat. no. 120237). 10,000 CD34+ cells from each donor were loaded on a GemCode Single-Cell Instrument (10x Genomics, Pleasanton, CA, USA) to generate single-cell Gel Bead in emulsion (GEMs) respectively. Reverse transcription (RT) was performed according to the manufacturer's protocol (55\\u2009°C for 2\\u2009h then 85\\u2009°C for 5\\u2009min). After RT, GEMs were broken and the single-strand cDNA was purified with DynaBeads MyOne Silane Beads and SPRIselect Reagent (0.6 × SPRI). cDNA was amplified by PCR (98\\u2009°C for 3\\u2009min; 98\\u2009°C for 15\\u2009s; 67\\u2009°C for 20\\u2009s; and 72\\u2009°C for 1\\u2009min) for 14 cycles then 72\\u2009°C for 1\\u2009min. The amplified cDNA product was cleaned up using SPRIselect beads (0.6 × SPRI). The cDNA was enzymatically fragmented and end-repair, dA-tailing and adaptor ligation were performed. The patient samples were differentially indexed using Chromium i7 sample indexing. The barcoded and indexed libraries were quantified by quantitative PCR (KAPA Biosystems Library Quantification Kit for Illumina platforms, cat. no. KK4824) and normalised. The library pool was sequenced on an Illumina NovaSeq6000 (S2 kits, PE100bp) (Novogene). Approximately 3,000-5,000 cells were captured for each donor and at least 100,000 reads per cell were obtained (QC30 shown in supplemental Table 2).\", 'The Cell Ranger Single-Cell Software (1.2.0, 10x Genomics) was used to perform sequencing data demultiplexing, barcode processing and gene counting against the reference genome (GRCh38-1.2.0).', 'Genome_build: GRCh38-1.2.0', 'Supplementary_files_format_and_content: tar archive containing genes.tsv, matrix.mtx, barcodes.tsv'], 'platform_id': ['GPL24676'], 'contact_name': ['Peng,,Hua'], 'contact_email': ['penghua@njmu.edu.cn'], 'contact_department': ['State Key Laboratory of Reproductive Medicine and Offspring Health'], 'contact_institute': ['Nanjing Medical University'], 'contact_address': ['101 Longmiandadao'], 'contact_city': ['Nanjing'], 'contact_state': ['Jiangsu'], 'contact_zip/postal_code': ['211116'], 'contact_country': ['China'], 'instrument_model': ['Illumina NovaSeq 6000'], 'library_selection': ['cDNA'], 'library_source': ['transcriptomic'], 'library_strategy': ['RNA-Seq'], 'relation': ['BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN12111242', 'SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX6261317'], 'supplementary_file_1': ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3901nnn/GSM3901498/suppl/GSM3901498_SCD5.tar.gz'], 'series_id': ['GSE133181'], 'data_row_count': ['0']}\n",
      "One row down\n"
     ]
    }
   ],
   "source": [
    "import GEOparse\n",
    "import pandas as pd\n",
    "\n",
    "# Replace 'GSE133181' with the desired GEO accession number\n",
    "geo = GEOparse.get_GEO('GSE133181')\n",
    "\n",
    "print(\"\\n\\n\")\n",
    "# Get basic information about the dataset\n",
    "print(\"basic info about data set:\")\n",
    "display(geo.metadata)\n",
    "\n",
    "\n",
    "# Access individual samples (GSMs)\n",
    "for gsm_name, gsm in geo.gsms.items():\n",
    "    # display(gsm_name)\n",
    "    # display(gsm.metadata)\n",
    "\n",
    "    print(gsm_name)\n",
    "    # print(gsm_name.title) # gets us built in metho of a string object\n",
    "    print(gsm.metadata)\n",
    "\n",
    "    print(\"One row down\")\n",
    "    # Access expression data, if available\n",
    "    # AttributeError: 'GSE' object has no attribute 'table'\n",
    "    # print(geo.table)\n",
    "\n",
    "\n",
    "# will this work?\n",
    "# nope\n",
    "# expression_data = geo.expression_data()\n",
    "\n",
    "# List names of columns in data\n",
    "# print(list(geo.columns))\n",
    "\n",
    "# geo.table\n",
    "\n",
    "# Access the expression data\n",
    "# expression_data = geo.pivot_table('VALUE', 'ID_REF', 'GSM')\n",
    "\n",
    "# expression_data_testing = geo.pivot_samples('VALUE', 'ID_REF')\n",
    "\n",
    "# Access sample information\n",
    "# sample_info = geo.phenotype_data\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "27-Nov-2024 12:35:31 DEBUG utils - Directory ./ already exists. Skipping.\n",
      "27-Nov-2024 12:35:31 INFO GEOparse - File already exist: using local version.\n",
      "27-Nov-2024 12:35:31 INFO GEOparse - Parsing ./GSE133181_family.soft.gz: \n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SERIES: GSE133181\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - PLATFORM: GPL24676\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901485\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901486\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901487\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901488\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901489\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901490\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901491\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901492\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901493\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901494\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901495\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901496\n",
      "27-Nov-2024 12:35:31 DEBUG GEOparse - SAMPLE: GSM3901498\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'GSE' object has no attribute 'expression_data'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[26], line 8\u001b[0m\n\u001b[0;32m      5\u001b[0m geo \u001b[38;5;241m=\u001b[39m GEOparse\u001b[38;5;241m.\u001b[39mget_GEO(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGSE133181\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m      7\u001b[0m \u001b[38;5;66;03m# Extract the expression data as a pandas DataFrame\u001b[39;00m\n\u001b[1;32m----> 8\u001b[0m expression_data \u001b[38;5;241m=\u001b[39m \u001b[43mgeo\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mexpression_data\u001b[49m\n\u001b[0;32m     10\u001b[0m \u001b[38;5;66;03m# Clean the data (example: handle missing values)\u001b[39;00m\n\u001b[0;32m     11\u001b[0m expression_data\u001b[38;5;241m.\u001b[39mfillna(\u001b[38;5;241m0\u001b[39m, inplace\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)  \u001b[38;5;66;03m# Replace missing values with 0\u001b[39;00m\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'GSE' object has no attribute 'expression_data'"
     ]
    }
   ],
   "source": [
    "import GEOparse\n",
    "import pandas as pd\n",
    "\n",
    "# Download the GEO dataset\n",
    "geo = GEOparse.get_GEO('GSE133181')\n",
    "\n",
    "# Extract the expression data as a pandas DataFrame\n",
    "expression_data = geo.expression_data\n",
    "\n",
    "# Clean the data (example: handle missing values)\n",
    "expression_data.fillna(0, inplace=True)  # Replace missing values with 0\n",
    "\n",
    "# Filter data (example: keep only genes with expression > 10 in at least 5 samples)\n",
    "filtered_data = expression_data.loc[:, (expression_data > 10).sum() >= 5]\n",
    "\n",
    "# Normalize data (example: log2 transformation)\n",
    "normalized_data = np.log2(filtered_data + 1)\n",
    "\n",
    "# Save the cleaned and preprocessed data\n",
    "normalized_data.to_csv('cleaned_data.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "sc-rna-sickle-cell",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
